homeassistant/components/sunricher_dali/entity.py

```diff
@@ -0,0 +1,53 @@
+"""Base entity for Sunricher DALI integration."""
+
+from __future__ import annotations
+
+import logging
+
+from PySrDaliGateway import CallbackEventType, DaliObjectBase, Device
+
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DaliCenterEntity(Entity):
+    """Base entity for DALI Center objects (devices, scenes, etc.)."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, dali_object: DaliObjectBase) -> None:
+        """Initialize base entity."""
+        self._dali_object = dali_object
+        self._attr_unique_id = dali_object.unique_id
+        self._unavailable_logged = False
+        # Set initial availability from status if available (Device has it, Scene doesn't)
+        if isinstance(dali_object, Device):
+            self._attr_available = dali_object.status == "online"
+        else:
+            self._attr_available = True
```

Comment:
You could consider just creating a second base entity that inherits `DaliCenterEntity` and call it `DaliCenterDeviceEntity`. This way you dont have to do an isinstance check